### ISSUE TITLE
Weight calculator on product variant

### ISSUE THREAD

**Issue by ylastapis** [Sylius/Sylius#6972]
Hi Sylius Community,

I am currently developing a new ecommerce website using Sylius and I am looking for guidance on how best to go about finding a solution to a shipping weight problem.

We have a Paper based product that has a weight that can vary depending on the format and paper type that the customer chooses.

We know the weight of the paper per m². 
We need to be able to calculate the correct weight and apply it to each product variant.

Maybe a solution like the Pricing Calculator would be an option ?
Do you have any ideas on where to begin?

Kind Regards

**Comment by michalmarcinkowski** [comment]
If you want to calculate shipping fee based on product weight you need to implement shipping calculator similar to [PerUnitRateCalculator](https://github.com/Sylius/Sylius/blob/master/src/Sylius/Component/Shipping/Calculator/PerUnitRateCalculator.php).

**Comment by ylastapis** [comment]
Thanks @michalmarcinkowski for your answer. I really like your approach with these calculators :).

That's the conclusion I've ended my searches.
I've created a DelegatingWeightCalculator which job is to compute weight of variants upon creation, depending of their archetype & product options.

As my code is very dependent of my business process, I don't think a PR will be interesting for community. But if anyone needs some help on this point, I could make a gist.
